libraries/AP_Frsky_Telem/AP_Frsky_D.cpp

```diff
@@ -49,7 +49,9 @@ void AP_Frsky_D::send(void)
         _D.last_200ms_frame = now;
         send_uint16(DATA_ID_TEMP2, (uint16_t)(AP::gps().num_sats() * 10 + AP::gps().status())); // send GPS status and number of satellites as num_sats*10 + status (to fit into a uint8_t)
         send_uint16(DATA_ID_TEMP1, gcs().custom_mode()); // send flight mode
-        send_uint16(DATA_ID_FUEL, (uint16_t)roundf(_battery.capacity_remaining_pct())); // send battery remaining
+        uint8_t percentage = 0;
+        IGNORE_RETURN(_battery.capacity_remaining_pct(percentage));
+        send_uint16(DATA_ID_FUEL, (uint16_t)roundf(percentage)); // send battery remaining
```

Comment:
Does this strike you as a bit weird?

We convert an *integer* to a float, round it, then cast it back into a uint16_t.

Similarly elsewhere.

Pre-existing, so not your problem to fix.